[BUG] Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunc

**Describe the bug**
integration tests failed in databricks runtimes (7.3, 8.2), tests added in https://github.com/NVIDIA/spark-rapids/pull/2596

```bash
[2021-06-14T15:18:35.077Z] 

[2021-06-14T15:18:35.077Z] e = IllegalArgumentException('Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunc...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:251)\n\tat java.lang.Thread.run(Thread.java:748)\n', None)

[2021-06-14T15:18:35.077Z] 

[2021-06-14T15:18:35.077Z] >   ???

[2021-06-14T15:18:35.077Z] E   pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunctionExec

[2021-06-14T15:18:35.077Z] E   RunningWindowFunction [row_number() windowspecdefinition(b#139233, a#139232L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS row_num#139241, count(c#139234L) windowspecdefinition(b#139233, a#139232L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS count_col#139242L, min(c#139234L) windowspecdefinition(b#139233, a#139232L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS min_col#139243L, max(c#139234L) windowspecdefinition(b#139233, a#139232L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS max_col#139244L], [b#139233], [a#139232L ASC NULLS FIRST]

[2021-06-14T15:18:35.077Z] E   +- GpuColumnarToRow false

[2021-06-14T15:18:35.077Z] E      +- GpuSort [b#139233 ASC NULLS FIRST, a#139232L ASC NULLS FIRST], false, com.nvidia.spark.rapids.OutOfCoreSort$@714f424c

[2021-06-14T15:18:35.077Z] E         +- GpuShuffleCoalesce 1000

[2021-06-14T15:18:35.077Z] E            +- GpuColumnarExchange gpuhashpartitioning(b#139233, 12), true, [id=#312973]

[2021-06-14T15:18:35.077Z] E               +- GpuProject [b#139233, a#139232L, c#139234L]

[2021-06-14T15:18:35.077Z] E                  +- GpuRowToColumnar targetsize(1000)

[2021-06-14T15:18:35.077Z] E                     +- Scan ExistingRDD[a#139232L,b#139233,c#139234L]

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z] <string>:3: IllegalArgumentException

[2021-06-14T15:18:35.078Z] ----------------------------- Captured stdout call -----------------------------

[2021-06-14T15:18:35.078Z] ### CPU RUN ###

[2021-06-14T15:18:35.078Z] ### GPU RUN ###

[2021-06-14T15:18:35.078Z] ______________________ test_window_running[Null-Long-1g] _______________________

[2021-06-14T15:18:35.078Z] [gw4] linux -- Python 3.7.10 /databricks/conda/envs/databricks-ml-gpu/bin/python

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z] b_gen = Null, c_gen = Long, batch_size = '1g'

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z]     @ignore_order

[2021-06-14T15:18:35.078Z]     @pytest.mark.parametrize('batch_size', ['1000', '1g'], ids=idfn) # set the batch size so we can test multiple stream batches

[2021-06-14T15:18:35.078Z]     @pytest.mark.parametrize('b_gen, c_gen', [(long_gen, x) for x in running_part_and_order_gens] +

[2021-06-14T15:18:35.078Z]             [(x, long_gen) for x in all_basic_gens_no_nans + [decimal_gen_scale_precision]], ids=idfn)

[2021-06-14T15:18:35.078Z]     def test_window_running(b_gen, c_gen, batch_size):

[2021-06-14T15:18:35.078Z]         conf = {'spark.rapids.sql.batchSizeBytes': batch_size,

[2021-06-14T15:18:35.078Z]                 'spark.rapids.sql.hasNans': False,

[2021-06-14T15:18:35.078Z]                 'spark.rapids.sql.castFloatToDecimal.enabled': True}

[2021-06-14T15:18:35.078Z]         query_parts = ['row_number() over (partition by b order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as row_num',

[2021-06-14T15:18:35.078Z]                 'count(c) over (partition by b order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as count_col',

[2021-06-14T15:18:35.078Z]                 'min(c) over (partition by b order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as min_col',

[2021-06-14T15:18:35.078Z]                 'max(c) over (partition by b order by a rows between UNBOUNDED PRECEDING AND CURRENT ROW) as max_col']

[2021-06-14T15:18:35.078Z]         assert_gpu_and_cpu_are_equal_sql(

[2021-06-14T15:18:35.078Z]             lambda spark : three_col_df(spark, LongRangeGen(), RepeatSeqGen(b_gen, length=100), c_gen, length=1024 * 14),

[2021-06-14T15:18:35.078Z]             "window_agg_table",

[2021-06-14T15:18:35.078Z]             'select ' +

[2021-06-14T15:18:35.078Z]             ', '.join(query_parts) +

[2021-06-14T15:18:35.078Z]             ' from window_agg_table ',

[2021-06-14T15:18:35.078Z] >           conf = conf)

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z] ../../src/main/python/window_function_test.py:322: 

[2021-06-14T15:18:35.078Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2021-06-14T15:18:35.078Z] ../../src/main/python/asserts.py:422: in assert_gpu_and_cpu_are_equal_sql

[2021-06-14T15:18:35.078Z]     assert_gpu_and_cpu_are_equal_collect(do_it_all, conf, is_cpu_first=is_cpu_first)

[2021-06-14T15:18:35.078Z] ../../src/main/python/asserts.py:386: in assert_gpu_and_cpu_are_equal_collect

[2021-06-14T15:18:35.078Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)

[2021-06-14T15:18:35.078Z] ../../src/main/python/asserts.py:367: in _assert_gpu_and_cpu_are_equal

[2021-06-14T15:18:35.078Z]     run_on_gpu()

[2021-06-14T15:18:35.078Z] ../../src/main/python/asserts.py:361: in run_on_gpu

[2021-06-14T15:18:35.078Z]     from_gpu = with_gpu_session(bring_back, conf=conf)

[2021-06-14T15:18:35.078Z] ../../src/main/python/spark_session.py:105: in with_gpu_session

[2021-06-14T15:18:35.078Z]     return with_spark_session(func, conf=copy)

[2021-06-14T15:18:35.078Z] ../../src/main/python/spark_session.py:70: in with_spark_session

[2021-06-14T15:18:35.078Z]     ret = func(_spark)

[2021-06-14T15:18:35.078Z] ../../src/main/python/asserts.py:189: in <lambda>

[2021-06-14T15:18:35.078Z]     bring_back = lambda spark: limit_func(spark).collect()

[2021-06-14T15:18:35.078Z] /databricks/spark/python/pyspark/sql/dataframe.py:611: in collect

[2021-06-14T15:18:35.078Z]     sock_info = self._jdf.collectToPython()

[2021-06-14T15:18:35.078Z] /databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py:1305: in __call__

[2021-06-14T15:18:35.078Z]     answer, self.gateway_client, self.target_id, self.name)

[2021-06-14T15:18:35.078Z] /databricks/spark/python/pyspark/sql/utils.py:133: in deco

[2021-06-14T15:18:35.078Z]     raise_from(converted)

[2021-06-14T15:18:35.078Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z] e = IllegalArgumentException('Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunc...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:251)\n\tat java.lang.Thread.run(Thread.java:748)\n', None)

[2021-06-14T15:18:35.078Z] 

[2021-06-14T15:18:35.078Z] >   ???

[2021-06-14T15:18:35.078Z] E   pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunctionExec

[2021-06-14T15:18:35.078Z] E   RunningWindowFunction [row_number() windowspecdefinition(b#139308, a#139307L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS row_num#139316, count(c#139309L) windowspecdefinition(b#139308, a#139307L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS count_col#139317L, min(c#139309L) windowspecdefinition(b#139308, a#139307L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS min_col#139318L, max(c#139309L) windowspecdefinition(b#139308, a#139307L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS max_col#139319L], [b#139308], [a#139307L ASC NULLS FIRST]

[2021-06-14T15:18:35.079Z] E   +- GpuColumnarToRow false

[2021-06-14T15:18:35.079Z] E      +- GpuSort [b#139308 ASC NULLS FIRST, a#139307L ASC NULLS FIRST], false, com.nvidia.spark.rapids.OutOfCoreSort$@714f424c

[2021-06-14T15:18:35.079Z] E         +- GpuShuffleCoalesce 1073741824

[2021-06-14T15:18:35.079Z] E            +- GpuColumnarExchange gpuhashpartitioning(b#139308, 12), true, [id=#313261]

[2021-06-14T15:18:35.079Z] E               +- GpuProject [b#139308, a#139307L, c#139309L]

[2021-06-14T15:18:35.079Z] E                  +- GpuRowToColumnar targetsize(1073741824)

[2021-06-14T15:18:35.079Z] E                     +- Scan ExistingRDD[a#139307L,b#139308,c#139309L]

[2021-06-14T15:18:35.079Z] 

[2021-06-14T15:18:35.079Z] <string>:3: IllegalArgumentException

[2021-06-14T15:18:35.079Z] ----------------------------- Captured stdout call -----------------------------

[2021-06-14T15:18:35.079Z] ### CPU RUN ###

[2021-06-14T15:18:35.079Z] ### GPU RUN ###

[2021-06-14T15:18:35.079Z] =============================== warnings summary ===============================

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] ../../../../../../databricks/spark/python/pyspark/sql/context.py:77

[2021-06-14T15:18:35.079Z] src/main/python/mortgage_test.py::test_mortgage[IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT, ALLOW_NON_GPU(ANY), LIMIT(100000)]

[2021-06-14T15:18:35.335Z] src/main/python/mortgage_test.py::test_mortgage[IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT, ALLOW_NON_GPU(ANY), LIMIT(100000)]

[2021-06-14T15:18:35.335Z]   /databricks/spark/python/pyspark/sql/context.py:77: DeprecationWarning: Deprecated in 3.0.0. Use SparkSession.builder.getOrCreate() instead.

[2021-06-14T15:18:35.335Z]     DeprecationWarning)

[2021-06-14T15:18:35.335Z] 

[2021-06-14T15:18:35.335Z] -- Docs: https://docs.pytest.org/en/stable/warnings.html

[2021-06-14T15:18:35.335Z] =========================== short test summary info ============================

[2021-06-14T15:18:35.335Z] FAILED ../../src/main/python/window_function_test.py::test_window_running_no_part[data:Null-1000][IGNORE_ORDER, APPROXIMATE_FLOAT]

[2021-06-14T15:18:35.335Z] FAILED ../../src/main/python/window_function_test.py::test_window_running_no_part[data:Null-1g][IGNORE_ORDER, APPROXIMATE_FLOAT]

[2021-06-14T15:18:35.335Z] FAILED ../../src/main/python/window_function_test.py::test_window_running[Null-Long-1000][IGNORE_ORDER]

[2021-06-14T15:18:35.335Z] FAILED ../../src/main/python/window_function_test.py::test_window_running[Null-Long-1g][IGNORE_ORDER]
```

--- a/shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/Spark301dbShims.scala
+++ b/shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/Spark301dbShims.scala
@@ -105,7 +105,7 @@ class Spark301dbShims extends Spark301Shims {
         "Databricks-specific window function exec, for \"running\" windows, " +
             "i.e. (UNBOUNDED PRECEDING TO CURRENT ROW)",
         ExecChecks(
-          TypeSig.commonCudfTypes + TypeSig.DECIMAL +
+          TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.Null +
               TypeSig.STRUCT.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL) +
               TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT
                   + TypeSig.ARRAY),

--- a/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/Spark311dbShims.scala
+++ b/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/Spark311dbShims.scala
@@ -108,7 +108,7 @@ class Spark311dbShims extends Spark311Shims {
         "Databricks-specific window function exec, for \"running\" windows, " +
             "i.e. (UNBOUNDED PRECEDING TO CURRENT ROW)",
         ExecChecks(
-            TypeSig.commonCudfTypes + TypeSig.DECIMAL +
+            TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.NULL +
                 TypeSig.STRUCT.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL) +
                 TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT
                     + TypeSig.ARRAY),

